.github/workflows/github-actions-tests.yml

```
@@ -15,27 +15,29 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.13t" ]
         mongodb-version: [ "4.4", "5.0", "6.0", "7.0", "8.0" ]
-        pydantic-version: [ "1.10.18", "2.9.2" , "2.10.4"]
+        pydantic-version: [ "1", "2"]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
```

If we look in the "run tests" stage output for every pipeline "run tests" job, we can see that the Python version used is always "Python 3.12.3", so the setting of the Python version is not working (since it was removed).
I've added back the "setup-python" stage:
```suggestion
      - uses: actions/checkout@v4
      - name: Set up Python ${{ inputs.python-version }}
        uses: actions/setup-python@v5
        with:
          python-version: ${{ inputs.python-version }}
```